### PR TITLE
Made FileIndex to implement ETL trait

### PR DESCRIPTION
**Pull request by botekchristophe**

### CODE REVIEW

File: src/main/scala/ca/cqdg/etl/FileIndex.scala [hunk 1]
```diff
@@ -61,21 +66,8 @@ object File {
...
-    //result.printSchema()
-    result
       .withColumn("dictionary_version", lit(studyNDF.dictionaryVersion))
       .withColumn("study_version", lit(studyNDF.studyVersion))
       .withColumn("study_version_creation_date", lit(studyNDF.studyVersionCreationDate))
   }
-
-  def write(files: DataFrame, outputPath: String)(implicit spark: SparkSession): Unit = {
-    files
-      .coalesce(1)
```
Comment (creativeyann17) [inline src/main/scala/ca/cqdg/etl/FileIndex.scala:75]: Is this the reason we now have several files in the output instead of 1 file with all the results ?

**Review by creativeyann17 — APPROVED** [review]
Not a lot of comments. Most of them would be me learning stuff about Scala and good practices.
- I suppose you did File as a PoC so we can do Study/Donor later ;)
- Some File joins could be Seq(...) but not blocking I think you don't want to take the risk of breaking the logic.
- Local run diff before/after in a comment below

File: src/main/scala/ca/cqdg/etl/FileIndex.scala (hunk 1 above)
Comment (botekchristophe) [reply src/main/scala/ca/cqdg/etl/FileIndex.scala:75]: probably, good catch
unless we absolutely need 1 and only 1 file for the ouput. it's faster to write multiple files on disk as the process is (supposed to be) executed in parallel

Comment (creativeyann17) [reply src/main/scala/ca/cqdg/etl/FileIndex.scala:75]: Ok, I will check if the indexer is good with that

**Pull request merged**